marklogic-data-hub/src/main/java/com/marklogic/hub/impl/ArtifactManagerImpl.java

```diff
@@ -29,18 +30,20 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
 
 @Component
 public class ArtifactManagerImpl implements ArtifactManager {
     HubConfig hubConfig;
+    protected List<ArtifactTypeInfo> allArtifactTypeInfo = null;
```

Comment:
What's the benefit of saving this state in the class? Are there use cases where we may invoke multiple methods on an instance of this class for the same user?

Reply:
No longer keeping the state. With the change to deliberately state whether the current user can update an artifact type, we want to ensure that information is current anyway.